### ISSUE TITLE
How do the results compare to other methods, such as U-Net?

### ISSUE THREAD

**Issue by chagelo** [Piogeon/InDI-implementation#3]
I run this code, but very slowly. One epoch cost almost one hour, batch_size 4 is too large on 3090, 24GB.

So I want to know the result is better or worse compare to other method, such as Unet. And my task is medical image denoising.

**Comment by fpramunno** [comment]
Hi, so for comparison you should read the paper and you can see https://arxiv.org/abs/2303.11435. Short answer it should be better than a simple Unet. For the speed, it depends by a lot of factors: image size, training dataset size, ... . In addition, an hint is to reduce the image_size parameters when you define the model, it is not strictly related to the image size of your image, but it controls the channels of the convolutional layers.

**Comment by chagelo** [comment]
Thank you! After modifying the model's image_size parameter, it works fine.
And I' am doing the supervised denoising using ddpm, but get worse result
than Unet.🥲


Francesco Pio Ramunno ***@***.***> 于2024年7月22日周一 16:28写道：

**Comment by fpramunno** [comment]
Which gets worse results, ddpm or indi?

**Comment by chagelo** [comment]
DDPM gets worse results, about 0.5 psnr less than Unet

Francesco Pio Ramunno ***@***.***> 于2024年7月22日周一 16:55写道：

**Comment by fpramunno** [comment]
Also here it can depends by multiple factors. Are you aware of the Perception–Distortion trade-off? PSNR and others pixel level metrics like SSIM tends to prefer blurry output (it is also stated in the Palette paper https://arxiv.org/abs/2111.05826), so use perceptual metrics like lpips or similar. In addition the model size should be conisdered as an hyperparameter lowering this parameter improves the speed but with a cost of output quality. A fair comparison would be to use the exact same Unet for the basic approac and the DDPM/INDI approach.

**Comment by chagelo** [comment]
Thank you for the suggestions, I'll calculate the perceptual metrics.
However, my dataset is medical images, I'm not sure if it's suitable to use
perceptual metric in medical image low-level vision tasks.

Francesco Pio Ramunno ***@***.***> 于2024年7月22日周一 17:07写道：

**Comment by fpramunno** [comment]
That's really dependent on your data, anyway it is always useful to have several metrics of different nature!

**Comment by chagelo** [comment]
OK. And very grateful.

Francesco Pio Ramunno ***@***.***> 于2024年7月22日周一 17:56写道：

**Comment by fpramunno** [comment]
No problem at all, for any question just ask! :)
